File: py-sdk/src/forta_bot/jwt/decode_jwt.py
import base64
from typing import Any, Callable, TypedDict

class DecodedJwt(TypedDict):
    header: Any
    payload: Any

DecodeJwt = Callable[[str], DecodedJwt]

def provide_decode_jwt() -> DecodeJwt:
    
  def decode_jwt(token):
      # Add 4 bytes for pythons b64decode
      header = base64.urlsafe_b64decode(
          token.split('.')[0] + '==').decode('utf-8')
      payload = base64.urlsafe_b64decode(
          token.split('.')[1] + '==').decode('utf-8')

      return {
          "header": header,
          "payload": payload
      }
  
  return decode_jwt

File: ts-sdk/src/jwt/decode.jwt.ts
export type DecodeJwt = (token: string) => DecodedJwt;

export interface DecodedJwt {
  header: any;
  payload: any;
}

export function provideDecodeJwt(): DecodeJwt {
  return function decodeJwt(token: string) {
    const splitJwt = token.split(".");
    const header = JSON.parse(Buffer.from(splitJwt[0], "base64").toString());
    const payload = JSON.parse(Buffer.from(splitJwt[1], "base64").toString());

    return {
      header,
      payload,
    };
  };
}
